base_rest/restapi.py

```diff
@@ -0,0 +1,250 @@
+
+    def to_openapi_requestbody(self, service):
+        return {}
+
+    def to_openapi_response(self, service):
+        return {}
+
+
+class CerberusValidator(RestMethodParam):
+    def __init__(self, schema=None, is_list=False):
+        """
+
+        :param schema: can be dict as cerberus schema, an instance of
+                       cerberus.Validator or a sting with the method name to
+                       call on the service to get the schema or the validator
+        :param is_list:
+        """
+        self._schema = schema
+        self._is_list = is_list
+
+    def from_params(self, service, params):
+        validator = self.get_cerberus_validator(service)
+        if validator.validate(params):
+            return validator.document
+        raise UserError(_("BadRequest %s") % validator.errors)
+
+    def to_response(self, service, result):
+        validator = self.get_cerberus_validator(service)
+        if validator.validate(result):
+            return validator.document
+        raise SystemError(_("Invalid Response %s") % validator.errors)
+
+    def to_openapi_query_parameters(self, service):
+        json_schema = self.to_json_schema(service)
+        parameters = []
+        for prop, spec in list(json_schema["properties"].items()):
+            params = {
+                "name": prop,
+                "in": "query",
+                "required": prop in json_schema["required"],
+                "allowEmptyValue": spec.get("nullable", False),
+                "default": spec.get("default"),
+            }
+            if spec.get("schema"):
+                params["schema"] = spec.get("schema")
+            else:
+                params["schema"] = {"type": spec["type"]}
+            if spec.get("items"):
+                params["schema"]["items"] = spec.get("items")
+            if "enum" in spec:
+                params["schema"]["enum"] = spec["enum"]
+
+            parameters.append(params)
+
+            if spec["type"] == "array":
+                # To correctly handle array into the url query string,
+                # the name must ends with []
+                params["name"] = params["name"] + "[]"
+
+        return parameters
+
+    def to_openapi_requestbody(self, service):
+        return {
+            "content": {"application/json": {"schema": self.to_json_schema(service)}}
+        }
+
+    def to_openapi_responses(self, service):
+        json_schema = self.to_json_schema(service)
+        return {"200": {"content": {"application/json": {"schema": json_schema}}}}
+
+    def get_cerberus_validator(self, service):
+        schema = None
+        if isinstance(self._schema, dict):
+            # schema is a cerberus schema
+            schema = self._schema
+        elif isinstance(self._schema, str):
+            # schema is a method name to call on service to get the schema or
+            schema = getattr(service, self._schema)()
+        if isinstance(schema, Validator):
+            return schema
```

Comment:
@etobella IMO the right fix should be:
```suggestion
        schema = self._schema
        if isinstance(self._schema, dict):
            # schema is a cerberus schema
            schema = self._schema
        elif isinstance(self._schema, str):
            # schema is a method name to call on service to get the schema or
            schema = getattr(service, self._schema)()
        if isinstance(schema, Validator):
            return schema
```
We must add a test to cover this case

Reply:
Good for me